plugin-system/plugin.d.ts

```diff
@@ -6,8 +6,12 @@ export default interface PluginBaseClass {
     el: HTMLElement;
     $emitter: NativeEventEmitter;
     _pluginName: String;
-    initialOptions: object;
-    options: object;
+    initialOptions: {
+	[key: string]: any;
```

Comment:
`any` is bad; could it also be an object?

or even better a proper type defined.

Reply:
But you have to use any here since you also want the user to use their own types here...
Another way might be to put a generic onto the `Plugin` class itself. Then it might be used like this:

```ts
export default class CustomStorefrontPlugin extends Plugin<{ option1: string; option2: CustomType | null }> {
  static options = {
    option1 = 'Default string value',
    option2 = null
},

  init() {...}
```